docfiles/docs.js

```diff
@@ -259,6 +260,93 @@ function renderSnippets() {
     });
 }
 
+function getCookieLang() {    
+    const cookiePropRegex = new RegExp(`${pxt.Util.escapeForRegex(pxt.Util.pxtLangCookieId)}=(.*?)(?:;|$)`)
+    const cookieValue = cookiePropRegex.exec(document.cookie);
+    return cookieValue && cookieValue[1] || null;
+}
+
+function setCookieLang(langId) {
+    if (!pxt.Util.allLanguages[langId]) {
+        return;
+    }
+    if (langId !== getCookieLang()) {
+        pxt.tickEvent(`menu.lang.setcookielang.${langId}`);
+        const expiration = new Date();
+        expiration.setTime(expiration.getTime() + (pxt.Util.langCookieExpirationDays * 24 * 60 * 60 * 1000));
+        document.cookie = `${pxt.Util.pxtLangCookieId}=${langId}; expires=${expiration.toUTCString()}`;
+    }
+}
+
+function languageOption(code) {
+    let locale = pxt.Util.allLanguages[code];
+
+    var headerEl = document.createElement('div');
+    headerEl.className = 'header';
+    headerEl.textContent = locale.localizedName;
+
+    var descriptionEl = document.createElement('div');
+    descriptionEl.classList.add('description', 'tall');
+    descriptionEl.textContent = locale.englishName;
+
+    var contentEl = document.createElement('div');
+    contentEl.className = 'content';
+    contentEl.appendChild(headerEl);
+    contentEl.appendChild(descriptionEl);
+
+    var cardEl = document.createElement('div');
+    cardEl.classList.add('ui','card','link','card-selected','langoption');
+    cardEl.dataset.lang = code;
+    cardEl.setAttribute('role', 'link');
+    cardEl.setAttribute('aria-label', locale.englishName);
+    cardEl.setAttribute('tab-index', '0');
+    cardEl.appendChild(contentEl);
+
+    return cardEl;
+}
+
+function setupLangPicker() {
+
+    let availableLocales = pxt.appTarget.appTheme.availableLocales;
+    let locales;
+    let initialLang = pxt.Util.normalizeLanguageCode(getCookieLang())[0];
+    if (pxt.appTarget.appTheme && availableLocales && pxt.appTarget.appTheme.selectLanguage) {
+
+        let localesContainer = document.querySelector("#availablelocales");
+        locales = availableLocales.map(e => languageOption(e));
+        locales.forEach(card => localesContainer.appendChild(card));
+   
+        document.querySelector("#langpicker").onclick = function() { 
+            $('.ui.modal').modal('show'); 
+        };
+
+        document.querySelector(".closeIcon").onclick = function() { 
+            $('.ui.modal').modal('hide'); 
+        };
+
+        let langOptions = document.querySelectorAll(".langoption");
```

Comment:
test in IE11

Reply:
Sure

Reply:
Not sure “let” is supported over there

Reply:
Correct, those are not supported. Will change across the file. Also will verify anonymous functions are not arrow functions.